### ISSUE TITLE
Wallpaper restore fails on CM12.1

### ISSUE THREAD

**Issue by jcharaoui** [jensstein/oandbackup#127]
Restoring a Wallpaper backup from CyanogenMod 11 (4.4.4) on to CM12.1 (5.1.1) doesn't work. Although no error messages are shown, the original wallpaper remains.

Furthermore, after this, the original wallpaper can't be changed anymore. Attempting to change it through the settings UI has no effect. To fix this problem I had to delete /data/system/users/0/wallpaper and /data/system/users/0/wallpaper_info.xml and reboot the device.

**Comment by jensstein** [comment]
well, i wouldn't say this is unexpected. restoring system files across different operating systems cannot really be done in a reliable way (without manually mapping the differences).
there are no errors reported because the restore operation presumably completed without errors - the problem is just that the correctly restored files are not usable by the new operating system.
but if you have a good way to handle this problem i would be very glad to hear it :)

**Comment by jcharaoui** [comment]
I don't know enough about the Android OS to provide a true fix, but perhaps oandbackup could show a warning message in the restore dialog when it detects a different operating system version.

In the meantime, I also found out that restoring Wifi Access Points across OS versions with oandbackup is also non functional.

**Comment by jensstein** [comment]
yeah, i have thought about that. but one problem would be the actual detection of a different system. there are three attributes which indicate versioning: versionCode and versionName of a single package and api level of whole system.
versionCode is not usable since it only refers to the package itself. versionName often contains the version number of the android system on system packages. it's still not usable though, because it isn't consistent. on one of my systems three system packages in a row read ```4.2.1```, ```4.4.4-eng```, ```2.0.002 (715abbf45a-30)``` respectively, while most of the rest read ```4.4.4-715abbf45a```. on another system most of them read ```6.0.1``` but quite a lot of the others read ```1.0```. 
and api level is not usable either because it doesn't say anything about systems from different vendors or developers (e.g. android 5 from samsung is not the same as cm12).

so it seems to me that the best solution is simply to know that restoring systems packages is never really guarantied to work. unless i'm missing something of course.
